### ISSUE TITLE
JSF/PrimeFaces 500 on `credit_compnay_bill_pharmacy.xhtml` — `ComponentNotFoundException` for `":"` referenced from `j_idt551:j_idt603:txtVal`

### ISSUE THREAD

**Issue by buddhika75** [hmislk/hmis#15924]
**Title:** JSF/PrimeFaces 500 on `credit_compnay_bill_pharmacy.xhtml` — `ComponentNotFoundException` for `":"` referenced from `j_idt551:j_idt603:txtVal`

**Summary**
A 500 error occurs when rendering the Pharmacy Credit Company Bill page. PrimeFaces raises `ComponentNotFoundException` because an Ajax search expression resolves to `":"` (empty/invalid), referenced from `j_idt551:j_idt603:txtVal`.

**Environment**

* Module / Page: `rh/faces/credit/credit_compnay_bill_pharmacy.xhtml`
* Request URL: `/rh/faces/credit/credit_compnay_bill_pharmacy.xhtml`
* Time (server): **2025-10-16 12:09:55**
* App Server: Payara/GlassFish (exact version unknown)
* JSF: (please fill)
* PrimeFaces: (please fill)
* Browser/OS: (please fill)
* User/Role: (please fill)

**Impact**
Page fails to render; user cannot process Pharmacy Credit Company Bills on this view.

**Steps to Reproduce**

1. Navigate to `/rh/faces/credit/credit_compnay_bill_pharmacy.xhtml`.
2. Trigger rendering of the component containing `j_idt551:j_idt603:txtVal` (e.g., load page or focus/blur key input).
3. Observe server error 500.

**Expected Result**
Page renders and Ajax behaviours update/render target components successfully.

**Actual Result**
Server responds with 500. PrimeFaces cannot resolve a component for the search expression `":"`, referenced from `j_idt551:j_idt603:txtVal`.

**Error Information (verbatim, for Support)**

* **Error Time:** 2025-10-16 12:09:55 (Server Time)
* **Request URL:** `/rh/faces/credit/credit_compnay_bill_pharmacy.xhtml`
* **Status Code:** 500
* **Exception Type:** `class javax.faces.component.search.ComponentNotFoundException`
* **Error Message:** `Cannot find component for expressions ":" referenced from "j_idt551:j_idt603:txtVal".`
* **Exception Details:** `javax.faces.component.search.ComponentNotFoundException: Cannot find component for expressions ":" referenced from "j_idt551:j_idt603:txtVal".`

**Complete Stack Trace**

```
javax.faces.component.search.ComponentNotFoundException: Cannot find component for expressions ":" referenced from "j_idt551:j_idt603:txtVal".
    at com.sun.faces.component.search.SearchExpressionHandlerImpl.resolveClientIds(SearchExpressionHandlerImpl.java:117)
    at org.primefaces.expression.SearchExpressionUtils.resolveClientIdsAsString(SearchExpressionUtils.java:132)
    at org.primefaces.util.AjaxRequestBuilder.addExpressions(AjaxRequestBuilder.java:156)
    at org.primefaces.util.AjaxRequestBuilder.update(AjaxRequestBuilder.java:148)
    at org.primefaces.behavior.ajax.AjaxBehaviorRenderer.getScript(AjaxBehaviorRenderer.java:106)
    at javax.faces.component.behavior.ClientBehaviorBase.getScript(ClientBehaviorBase.java:80)
    at org.primefaces.renderkit.CoreRenderer.renderDomEvents(CoreRenderer.java:206)
    at org.primefaces.renderkit.CoreRenderer.renderDomEvents(CoreRenderer.java:134)
    at org.primefaces.component.inputtext.InputTextRenderer.encodeMarkup(InputTextRenderer.java:113)
    at org.primefaces.component.inputtext.InputTextRenderer.encodeEnd(InputTextRenderer.java:67)
    at javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:595)
    at com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:286)
    at com.sun.faces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:90)
    at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:566)
    at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1647)
    at com.sun.faces.renderkit.html_basic.CompositeRenderer.encodeChildren(CompositeRenderer.java:54)
    at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:566)
    at com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:278)
    at org.primefaces.renderkit.CoreRenderer.renderChild(CoreRenderer.java:82)
    at org.primefaces.renderkit.CoreRenderer.renderChildren(CoreRenderer.java:69)
    at org.primefaces.component.panel.PanelRenderer.encodeContent(PanelRenderer.java:243)
    at org.primefaces.component.panel.PanelRenderer.encodeMarkup(PanelRenderer.java:149)
    at org.primefaces.component.panel.PanelRenderer.encodeEnd(PanelRenderer.java:76)
    at javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:595)
    at org.primefaces.renderkit.CoreRenderer.renderChild(CoreRenderer.java:87)
    at org.primefaces.renderkit.CoreRenderer.renderChildren(CoreRenderer.java:69)
    at org.primefaces.component.panel.PanelRenderer.encodeContent(PanelRenderer.java:243)
    at org.primefaces.component.panel.PanelRenderer.encodeMarkup(PanelRenderer.java:149)
    at org.primefaces.component.panel.PanelRenderer.encodeEnd(PanelRenderer.java:76)
    at javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:595)
    at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1654)
    at javax.faces.render.Renderer.encodeChildren(Renderer.java:152)
    at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:566)
    at com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:278)
    at com.sun.faces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:90)
    at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:566)
    at org.primefaces.renderkit.CoreRenderer.renderChild(CoreRenderer.java:82)
    at org.primefaces.renderkit.CoreRenderer.renderChildren(CoreRenderer.java:69)
    at org.primefaces.component.panel.PanelRenderer.encodeContent(PanelRenderer.java:243)
    at org.primefaces.component.panel.PanelRenderer.encodeMarkup(PanelRenderer.java:149)
    at org.primefaces.component.panel.PanelRenderer.encodeEnd(PanelRenderer.java:76)
    at javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:595)
    at com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:286)
    at org.primefaces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:90)
    at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:566)
    at org.primefaces.renderkit.CoreRenderer.renderChild(CoreRenderer.java:82)
    at org.primefaces.renderkit.CoreRenderer.renderChildren(CoreRenderer.java:69)
    at org.primefaces.component.panel.PanelRenderer.encodeContent(PanelRenderer.java:243)
    at org.primefaces.component.panel.PanelRenderer.encodeMarkup(PanelRenderer.java:149)
    at org.primefaces.component.panel.PanelRenderer.encodeEnd(PanelRenderer.java:76)
    at javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:595)
    at com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:286)
    at org.primefaces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:90)
    at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:566)
    at com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:278)
    at org.primefaces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:90)
    at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:566)
    at com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:278)
    at org.primefaces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:90)
    at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:566)
    at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1647)
    at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1650)
    at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1650)
    at com.sun.faces.application.view.FaceletViewHandlingStrategy.renderView(FaceletViewHandlingStrategy.java:468)
    at com.sun.faces.application.view.MultiViewHandler.renderView(MultiViewHandler.java:170)
    at javax.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:132)
    at com.sun.faces.lifecycle.RenderResponsePhase.execute(RenderResponsePhase.java:102)
    at com.sun.faces.lifecycle.Phase.doPhase(Phase.java:76)
    at com.sun.faces.lifecycle.LifecycleImpl.render(LifecycleImpl.java:199)
    at javax.faces.webapp.FacesServlet.executeLifecyle(FacesServlet.java:708)
    at javax.faces.webapp.FacesServlet.service(FacesServlet.java:451)
    at org.apache.catalina.core.StandardWrapper.service(StandardWrapper.java:1637)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:331)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:211)
    at org.glassfish.tyrus.servlet.TyrusServletFilter.doFilter(TyrusServletFilter.java:282)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:253)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:211)
    at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:257)
    at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:167)
    at org.apache.catalina.core.StandardPipeline.doInvoke(StandardPipeline.java:757)
    at org.apache.catalina.core.StandardPipeline.invoke(StandardPipeline.java:577)
    at com.sun.enterprise.web.WebPipeline.invoke(WebPipeline.java:99)
    at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:158)
    at org.apache.catalina.connector.CoyoteAdapter.doService(CoyoteAdapter.java:372)
    at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:239)
    at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:520)
    at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:217)
    at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:182)
    at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:156)
    at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:201)
    at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:95)
    at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:260)
    at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:177)
    at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:109)
    at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:88)
    at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:53)
```

**Suspected Cause**
A component uses a PrimeFaces/JSF Ajax attribute (`update`/`process`/`for`) that resolves to an empty search expression `":"`. This likely comes from a binding like `update=":"` or a computed value that ends up empty. The failing reference originates from `j_idt551:j_idt603:txtVal` within the page/panel hierarchy.

**Where to Look**

* `credit_compnay_bill_pharmacy.xhtml`: find any `<p:ajax>`, `update`, `process`, `for`, or `render` attributes near the `txtVal` input or its parent forms/panels.
* Any composite/component includes that set `update`/`process` dynamically (EL producing `":"`).
* Check if `@form`, `@this`, `@parent`, or concrete clientIds are more appropriate than an empty expression.

**Proposed Fix**
Replace invalid `update=":"` (or equivalent) with valid targets, e.g., `update="@form"` or a concrete clientId. Ensure any EL expressions never evaluate to an empty expression. Validate all Ajax behaviours under `txtVal`.

**Workarounds**
None identified. Page is blocked until the invalid search expression is corrected.

**Related Files/Refs**

* `rh/faces/credit/credit_compnay_bill_pharmacy.xhtml`
* Any included templates or composite components used by this page.

**Labels**
`area:pharmacy`, `type:bug`, `severity:high`, `primefaces`, `jsf`, `server-error-500`

**Assignee**
(please assign)

**Additional Notes**
Please copy **all** information above (not screenshots) into the issue for complete context.
